yt_dlp/extractor/bilibili.py

```diff
@@ -892,19 +892,29 @@ def _get_subtitles(self, *, ep_id=None, aid=None):
                 'aid': aid,
             })) or {}
         subtitles = {}
-        for sub in sub_json.get('subtitles') or []:
-            sub_url = sub.get('url')
-            if not sub_url:
-                continue
-            sub_data = self._download_json(
-                sub_url, ep_id or aid, errnote='Unable to download subtitles', fatal=False,
-                note='Downloading subtitles%s' % f' for {sub["lang"]}' if sub.get('lang') else '')
-            if not sub_data:
-                continue
-            subtitles.setdefault(sub.get('lang_key', 'en'), []).append({
-                'ext': 'srt',
-                'data': self.json2srt(sub_data)
-            })
+
+        for sub in traverse_obj(sub_json, ('video_subtitle', ..., {dict})):
+            srt_sub = sub.get('srt')
+            ass_sub = sub.get('ass')
+            lang_key = sub.get('lang_key', 'en')
+            sub_formats = subtitles.setdefault(lang_key, [])
+
+            if srt_sub:
+                sub_data = self._download_json(
+                    srt_sub.get('url'), ep_id or aid, errnote='Unable to download subtitles', fatal=False,
+                    note='Downloading subtitles%s' % f' for {sub["lang"]} ({lang_key})' if sub.get('lang') else '')
```

Comment:
```suggestion
                    note='Downloading subtitles%s ({lang_key})' % f' for {sub["lang"]}' if sub.get('lang') else '')
```
Since the language key is always known, we can also always include it